semigroups.cabal

```diff
@@ -128,6 +136,9 @@ library
     if flag(bytestring)
       build-depends: bytestring >= 0.9 && < 1
 
+    if flag(bytestring-builder) && impl(ghc < 7.8)
```

Comment:
Cabal experts (not myself) came up with the method used in [bytestring-builder](http://hackage.haskell.org/package/bytestring-builder-0.10.8.1.0/bytestring-builder.cabal).   You declare a flag,  and then allow cabal to figure out the right setting for the flag, which is reminiscent of logic programming.  I can't say I'm a big fan of this approach,  but it's the approach that seems to have been adopted by the community.